### DISCUSSION TITLE
How to display API response from action

### DISCUSSION THREAD

**Discussion by micebrain** [Budibase/budibase#3918] [Help]
Hello and Happy New Year!
I have form that is submitted with API call using the "Execute Query" action on the save button.
However in some cases the API  server returns an error message. All I see in the BB app is a red toaster with the message "An error occurred". Is there a way to:
1. Read the API response 
2. Ability to display a more informative response to user.

Thanks!

**Comment by mjashanks** [comment]
Hey @micebrain 

At the moment, neither are possible, although I believe that we will have No.1 complete this month - i.e. read the response from Execute Query.

For No.2 what would you expect here? Do you want to display information contained in the error response, or is it enough to customise the "an Error has occurred" message?

**Reply by micebrain** [reply]
Hi @mjashanks 
Thanks for the update on No. 1
For No.2, yes, I would like to display the the message from the error response. For example an optional binding would be good.
Consider an error response like

```
{
    "status": "error",
    "message": "Code entered has expired"
}
```

It would be great to have an optional binding where JS can then be used to extract the appropriate key to display.
